src/exec/builtins/functions.rs

```diff
@@ -84,26 +100,163 @@ pub fn list_directory(shell: &mut ShellState, args: Vec<&str>) -> Result<()> {
+            } else {
+                print!("{}  ", i.clone().green())
+            }
+        }
 
-    for file in files {
-        println!("{}", &file);
+        for i in &files {
+            if i.starts_with('.') {
+                print!("{}  ", i.clone().grey())
+            } else {
+                print!("{}  ", i.clone().white())
+            }
+
+        }
+
+        println!();
+
+        return Ok(());
     }
 
+    directories.append(&mut files);
+
+    let permission_view = match octal_permissions {
+        true => DirectoryListPermissionMode::Octal,
+        false => DirectoryListPermissionMode::String,
+    };
+
+    list_directory_long(directories, permission_view, path_to_read.clone(), permission_seperator);
+
     Ok(())
 }
 
+fn list_directory_long(item: Vec<String>, permission_format: DirectoryListPermissionMode, cwd: PathBuf, permission_seperator: bool) {
+    let path_to_read = cwd;
+    let mut file_size_len: usize = 0;
+    let mut file_size_len_last: usize = 0;
+    let mut username_len: usize = 0;
+    let mut username_len_last: usize = 0;
+
+    for i in &item {
+        let path = path_to_read.join(i);
+        let file_size = std::fs::metadata(path.to_path_buf()).unwrap().size();
+        let formatted_fsize = Size::from_bytes(file_size).to_string();
+
+        file_size_len_last = formatted_fsize.len();
+
+        if file_size_len_last > file_size_len {
+                file_size_len = file_size_len_last;
+        }
+    }
+
+    for i in &item {
+        let path = path_to_read.join(i);
+        
+        username_len_last = path.owner().unwrap().to_string().len();
+
+        if username_len_last > username_len {
+            username_len = username_len_last;
+        }
+    }
+
+    for i in &item {
+        let path = path_to_read.join(i);
+        let permission_octal =  { 
+            let x = format!("{:o}", std::fs::metadata(path.to_path_buf()).unwrap().permissions().mode()); 
+            x[x.len() - 3..].to_string() 
+        };
+
+        let permissions = match permission_format {
+            DirectoryListPermissionMode::Octal => {
+                permission_octal.white()
+            },
+                
+            DirectoryListPermissionMode::String => {
+                let permission_str = permission_octal;
+                let mut result = String::new();
+
+                let dash = "-".dark_grey();
+                let r = "r".white();
+                let w = "w".white();
+                let x = "x".white();
+
+                let mapping = vec![
+                    format!("{0}{0}{0}", dash),
+                    format!("{0}{0}{1}", dash, x), 
+                    format!("{0}{1}{0}", dash, w), 
+                    format!("{}{}{}", dash, w, x),
+                    format!("{}{1}{1}", r, dash),
+                    format!("{}{}{}", r, dash, x),
+                    format!("{}{}{}", r, w, dash),
+                    format!("{}{}{}", r, w, x)
```

Comment:
A lot of these could also be simplified to inline format:
```suggestion
                    format!("{r}{w}{x}")
```

Reply:
Fixed via b7f738c